resources/js/store/store-supporter-tag.tsx

```diff
@@ -0,0 +1,316 @@
+                    <div className='store-slider__subtext'>{trans('store.discount', { percent: this.discount })}</div>
+                  </div>
+                </div>
+              </div>
+            </div>
+            <div className='store-slider__presets'>
+              <span className='store-slider__presets-blurb'>{trans('supporter_tag.months')}</span>
+              {monthPresets.map((preset) => (
+                // TODO: button
+                <div
+                  key={preset}
+                  className={classWithModifiers('store-slider__preset', {
+                    active: this.duration >= preset,
+                    disabled: !this.isValidUser,
+                  })}
+                  data-months={preset}
+                  onClick={this.handlePresetClick}
+                >
+                  {preset}
+                </div>
+              ))}
+            </div>
+          </div>
+        </div>
+      </>
+    );
+  }
+
+  @action
+  private readonly getUser = (username: string) => {
+    this.xhr = $.ajax({
+      data: { username },
+      dataType: 'json',
+      type: 'POST',
+      url: route('users.check-username-exists'),
+    });
+
+    this.xhr
+      .done((data) => runInAction(() => {
+        this.user = data;
+      }))
+      .fail(onError)
+      .always(() => {
+        this.xhr = null;
+      });
+  };
+
+  private readonly handleBeforeCache = () => {
+    storeJson(jsonId, {
+      giftMessage: this.giftMessageRef.current?.value ?? '',
+      sliderValue: this.sliderValue,
+      username: this.username,
+    });
+  };
+
+  private readonly handlePresetClick = (event: React.SyntheticEvent<HTMLElement>) => {
+    const price = durationToPrice(+(event.currentTarget.dataset?.months ?? 0));
+    if (price != null && this.sliderRef.current != null) {
+      $(this.sliderRef.current).slider('value', sliderValueFrom(price));
+    }
+  };
+
+  @action
+  private readonly handleSliderValueChanged = (_event: JQueryEventObject, ui: SliderUIParams) => {
+    if (ui.value == null) return;
+    this.sliderValue = ui.value;
+  };
+
+  @action
+  private readonly handleUsernameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.username = event.currentTarget.value;
+
+    this.debouncedGetUser.cancel();
+    this.xhr?.abort();
+    this.xhr = null;
+
+    // reset to current user on empty
+    if (this.username === '') {
+      this.user = core.currentUserOrFail;
+    } else {
+      this.user = null;
+      this.debouncedGetUser(this.username);
+    }
+  };
+
+  private initializeSlider() {
+    const slider = this.sliderRef.current;
+    if (slider == null) return;
+
+    $(slider).find('.ui-slider-range').remove();
+
+    return $(slider).slider({
+      animate: true,
+      change: this.handleSliderValueChanged,
+      max: sliderValueFrom(maxValue),
+      min: sliderValueFrom(minValue),
+      range: 'min',
+      slide: this.handleSliderValueChanged,
+      step: 1,
```

Comment:
apparently `sliderValueFrom` isn't needed if `step` is set to something like 0.1? (or 0.125 to copy the current "resolution")